<?xml version="1.0" ?>

<launch>
      <!-- Enable rviz visualization -->
      <arg name="rviz"        default="true" />
      <!-- Set nodes on debug mode -->
      <arg name="debug"       default="false" />

      <!-- The URDF model of the robot -->
      <arg name="urdfFile"    value="$(find robot_description)/airbot/urdf/airbot_play.urdf" />
      <!-- The task file for the mpc. -->
      <arg name="taskFile"    value="$(find ocs2_airbot)/config/task.info" />
      <!-- The library folder to generate CppAD codegen into -->
      <arg name="libFolder"   value="$(find ocs2_airbot)/auto_generated/airbot" />
      
      <param name="taskFile"   value="$(arg taskFile)" />
      <param name="urdfFile"   value="$(arg urdfFile)" />
      <param name="libFolder"  value="$(arg libFolder)" />

      <group if="$(arg rviz)">
            <include file="$(find ocs2_mobile_manipulator_ros)/launch/include/visualize.launch">
                  <arg name="urdfFile" value="$(arg urdfFile)" />
            </include>
      </group>

      <node if="$(arg debug)" pkg="ocs2_airbot" type="airbot_mpc_mrt_node" name="airbot_mpc_mrt_node" 
            output="screen" launch-prefix="gnome-terminal -- gdb -ex run --args" />
      <node unless="$(arg debug)" pkg="ocs2_airbot" type="airbot_mpc_mrt_node" name="airbot_mpc_mrt_node" 
            output="screen" launch-prefix="" />

      <node if="$(arg rviz)" pkg="ocs2_mobile_manipulator_ros" type="mobile_manipulator_target" name="mobile_manipulator_target" 
            output="screen" launch-prefix="" />


</launch>